### ISSUE TITLE
reportPrivateImportUsage confused by library source

### ISSUE THREAD

**Issue by GBeauregard** [microsoft/pyright#2392]
**Describe the bug**
The Library `rich` (`pip install rich`) currently generates the reportPrivateImportUsage error:

`/home/gregory/Downloads/testcase/pytest/ricchy.py:5:16 - error: "Live" is not exported from module "rich.box.live" (reportPrivateImportUsage)`

but this error gets the source library wrong in case other libraries in the namespace were imported first. The expected error is from module "rich.live" and not "rich.box.live".

**To Reproduce**
```python
import rich.box
import rich.live


with rich.live.Live("hi!"):
    ...
```

**VS Code extension or command-line**
pyright 1.1.175 commandline Linux

**Additional context**
I'd like to help get the library in line. There's an issue tracking the problem for tables in the library, and as I mentioned in the bug introducing this error I believe it's due to spurious usage in  `__main__.py`, but I'm actually not sure why this error is triggering on Live in the library at all as I couldn't find a usage pattern matching the documentation. It isn't related to the bug, but I would appreciate any assistance if you have the time.

**Comment by erictraut** [comment]
This is included in pyright 1.1.176, which I just published. It will also be included in the next release of pylance.

**Comment by GBeauregard** [comment]
While 1.1.176 did fix the other issues (thanks!), it did not resolve the title issue for me (nor the sample).

```console
~/Documents/projects/pyffstream/src/pyffstream main*
pyffstream ❯ pyright --version
npx: installed 1 in 2.104s
pyright 1.1.176

~/Documents/projects/pyffstream/src/pyffstream main*
pyffstream ❯ pyright ~/Downloads/testcase/pytest/ricchy.py
npx: installed 1 in 1.985s
No configuration file found.
pyproject.toml file found at /home/gregory/Documents/projects/pyffstream.
Loading pyproject.toml file at /home/gregory/Documents/projects/pyffstream/pyproject.toml
Assuming Python platform Linux
Auto-excluding **/node_modules
Auto-excluding **/__pycache__
Auto-excluding .git
stubPath /home/gregory/Documents/projects/pyffstream/typings is not a valid directory.
Searching for source files
File or directory "/home/gregory/Documents/projects/pyffstream/src/pyffstream/--" does not exist.
Found 1 source file
/home/gregory/Downloads/testcase/pytest/ricchy.py
  /home/gregory/Downloads/testcase/pytest/ricchy.py:5:16 - error: "Live" is not exported from module "rich.box.live" (reportPrivateImportUsage)
1 error, 0 warnings, 0 infos
Completed in 0.72sec
```
